caffe2/operators/segment_reduction_op.cc

```diff
@@ -2,6 +2,47 @@
 
 namespace caffe2 {
 
+OpSchema::Cost CostInferenceForSparseLengths(
+    const OperatorDef& def,
+    const vector<TensorShape>& inputs,
+    bool use_weight) {
+  int min_num_of_inputs = 3 + use_weight;
+  CAFFE_ENFORCE_GE(
+      inputs.size(),
+      min_num_of_inputs,
+      def.type() + " requires at least " +
+          caffe2::to_string(min_num_of_inputs));
+
+  const TensorShape data = inputs[0];
+  const TensorShape indices = inputs[1 + use_weight];
+  const TensorShape lengths = inputs[2 + use_weight];
+
+  OpSchema::Cost c;
+  CAFFE_ENFORCE_GT(data.dims_size(), 0, "data requires at least 1 dimension");
+  uint64_t N = data.dims(0);
+  if (N == 0) {
+    return c;
+  }
+  uint64_t D = nElemFromDim(data) / N;
```

Comment:
why not do from dim 1 and then not divide by N ? The function name is misleading, it should really accept dim as the second argument

Reply:
I wasn't able to find a function that computes size from a certain dimension. You're right we should change nElemFromDim function. Can we do that in a separate PR? :)